### PR TITLE
Update to latest `pyprotoc-plugin`

### PR DESCRIPTION
**Pull request by CodingCanuck**
This required upgrading from bazel 5.0.0 to 5.1.0 to pick up the
[new starlark strings functions](https://blog.bazel.build/2022/03/24/bazel-5.1.html#starlark).

### CODE REVIEW

**Comment by CodingCanuck** [comment]
Formatting fixes are in https://github.com/reboot-dev/pyprotoc-plugin/pull/16

**Comment by rjhuijsman** [comment]
Looks like this also introduces a test service? Is that intentional?

It also looks like this uncovers an eventuals-related build breakage? Is there an issue for that yet?

**Comment by CodingCanuck** [comment]
TL;DR let's do a quick sync after your meeting to catch you up on the cross-repo juggling here.

> Looks like this also introduces a test service? Is that intentional?
> 
> It also looks like this uncovers an eventuals-related build breakage? Is there an issue for that yet?

No to all of the above.

I'm currently basing this PR on top of https://github.com/3rdparty/eventuals/pull/301 : *that* PR adds a test service, that's intentional, and it fails to build for reasons being discussed in that other PR.

This PR modifies the call to `cc_eventuals_library` used by the test service (this is to handle the pyprotoc-plugin API change I made in https://github.com/reboot-dev/pyprotoc-plugin/pull/13 : this was expected. It's part of https://github.com/reboot-dev/respect-api/issues/17 . I didn't file an issue to fix the break this will cause in other repos by updating them to this new format: I could file one issue per repo, but we'd need to file 4-5 issues [see list](https://github.com/reboot-dev/respect/issues/266#issuecomment-1086542590) and that doesn't seem high ROI)

This call to `cc_eventuals_library` doesn't exist at HEAD / without PR 301: once this PR is approved, I'll just rebase it onto main if 301 isn't approved yet.

**Review by while-false — APPROVED** [review]
LGTM! :tada:

**Comment by CodingCanuck** [comment]
> TL;DR let's do a quick sync after your meeting to catch you up on the cross-repo juggling here.
> 
> > Looks like this also introduces a test service? Is that intentional?
> > It also looks like this uncovers an eventuals-related build breakage? Is there an issue for that yet?
> 
> No to all of the above.
> 
> I'm currently basing this PR on top of #301 : _that_ PR adds a test service, that's intentional, and it fails to build for reasons being discussed in that other PR.
> 
> This PR modifies the call to `cc_eventuals_library` used by the test service (this is to handle the pyprotoc-plugin API change I made in [reboot-dev/pyprotoc-plugin#13](https://github.com/reboot-dev/pyprotoc-plugin/pull/13) : this was expected. It's part of [reboot-dev/respect-api#17](https://github.com/reboot-dev/respect-api/issues/17) . I didn't file an issue to fix the break this will cause in other repos by updating them to this new format: I could file one issue per repo, but we'd need to file 4-5 issues [see list](https://github.com/reboot-dev/respect/issues/266#issuecomment-1086542590) and that doesn't seem high ROI)
> 
> This call to `cc_eventuals_library` doesn't exist at HEAD / without PR 301: once this PR is approved, I'll just rebase it onto main if 301 isn't approved yet.

https://github.com/3rdparty/eventuals/pull/301 is still pending: rebasing onto main and merging with @while-false 's LGTM
